src/features/activity/ActivityDetailsScreen.tsx

```diff
@@ -0,0 +1,297 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import { Linking, ScrollView } from 'react-native'
+import { ConfirmedSignatureInfo } from '@solana/web3.js'
+import { useTranslation } from 'react-i18next'
+import Clipboard from '@react-native-community/clipboard'
+import Toast from 'react-native-simple-toast'
+import Animated from 'react-native-reanimated'
+import ListItem from '../../components/ListItem'
+import { EnrichedTransaction } from '../../types/solana'
+import ImageBox from '../../components/ImageBox'
+import BackScreen from '../../components/BackScreen'
+import Box from '../../components/Box'
+import Text from '../../components/Text'
+import AddressActivityItem from './AddressActivityItem'
+import ButtonPressable from '../../components/ButtonPressable'
+import CheckmarkFilled from '../../assets/images/checkmarkFill.svg'
+import { useColors } from '../../theme/themeHooks'
+import { ellipsizeAddress, solAddressIsValid } from '../../utils/accountUtils'
+import Error from '../../assets/images/error.svg'
+import { ActivityStackParamList } from './activityTypes'
+import BlurActionSheet from '../../components/BlurActionSheet'
+import useHaptic from '../../utils/useHaptic'
+import globalStyles from '../../theme/globalStyles'
+import { DelayedFadeIn } from '../../components/FadeInOut'
+import { useAppStorage } from '../../storage/AppStorageProvider'
+
+type Route = RouteProp<ActivityStackParamList, 'ActivityDetailsScreen'>
+
+const ActivityDetailsScreen = () => {
+  const route = useRoute<Route>()
+  const colors = useColors()
+  const { solanaNetwork: cluster } = useAppStorage()
+  const { t, i18n } = useTranslation()
+  const { triggerNavHaptic } = useHaptic()
+
+  const { transaction } = route.params
+
+  const [optionsOpen, setOptionsOpen] = useState(false)
+  const [selectedAddress, setSelectedAddress] = useState<string>('')
+
+  const dateLabel = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    const date = new Date()
+
+    if (enrichedTx.timestamp) {
+      date.setTime(enrichedTx.timestamp * 1000)
+    }
+
+    if (confirmedSig.blockTime) {
+      date.setTime(confirmedSig.blockTime * 1000)
+    }
+
+    // Format date in DD/MMMM/YYYY format
+    const formattedString = date.toLocaleDateString('en-US', {
+      day: 'numeric',
+      month: 'long',
+      year: 'numeric',
+    })
+
+    const time = `${date.getHours()}:${
+      date.getMinutes() > 9 ? date.getMinutes() : `0${date.getMinutes()}`
+    }`
+
+    return `${formattedString} | ${time}`
+  }, [transaction])
+
+  const activityImage = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    if (confirmedSig.err) {
+      return <Error color={colors.error} width={150} height={150} />
+    }
+
+    const { tokenTransfers } = enrichedTx
+
+    if (
```

Comment:
I think this would break in the case that `tokenTransfers` is an empty array. Could simplify to something like this...
```ts
      tokenTransfers?.length &&
      tokenTransfers[0].tokenMetadata?.json?.image

Reply:
Solid point. Let me refactor some of this logic

Reply:
You could reduce this 
```ts
      tokenTransfers[0].tokenMetadata &&
       tokenTransfers[0].tokenMetadata.json &&
       tokenTransfers[0].tokenMetadata.json.image
 ```
down to 
```ts
tokenTransfers[0].tokenMetadata?.json?.image
```

Reply:
done